src/jobservice/hook/hook_client.go

```diff
@@ -54,6 +55,13 @@ func NewClient(ctx context.Context) Client {
 		ExpectContinueTimeout: 1 * time.Second,
 		Proxy:                 http.ProxyFromEnvironment,
 	}
+	if commonhttp.InternalTLSEnabled() {
+		tlsConfig, err := commonhttp.GetInternalTLSConfig()
+		if err != nil {
+			panic(err)
```

Comment:
panic???

Reply:
yes, if you enabled tls but not provided cert file correctly. service should fail immediately.

Reply:
but it's the client, is it called when the service bootstraps?